### PR TITLE
Update AbstractDb.php

### PR DESCRIPTION
**Pull request by andrey-mokhov**
When RecordExists and NoRecordExists validators are used two and more times, system throws exception:
SQLSTATE[HY093]: Invalid parameter number: :where1

This error is related with creation query string with parameters where2, where3, etc with every calling prepareStatementForSqlObject method.
This solution lets use RecordExists and NoRecordExists validators two and more times.

### CODE REVIEW

**Comment by Maks3w** [comment]
Need a test

File: src/Db/AbstractDb.php [hunk 1]
```diff
@@ -43,6 +44,11 @@
      * @var Select
      */
     protected $select;
+    
+    /**
+     * @var StatementInterface
+     */
+    protected $statement;
```
Comment (Maks3w) [inline src/Db/AbstractDb.php]: change to private

File: src/Db/AbstractDb.php [hunk 2]
```diff
@@ -43,6 +44,11 @@
      * @var Select
      */
     protected $select;
+    
+    /**
+     * @var StatementInterface
```
Comment (Maks3w) [inline src/Db/AbstractDb.php]: StreamInterface|null

File: src/Db/AbstractDb.php [hunk 3]
```diff
@@ -190,6 +196,7 @@ public function setExclude($exclude)
     {
         $this->exclude = $exclude;
         $this->select  = null;
+        $this->statement = null;
```
Comment (Maks3w) [inline src/Db/AbstractDb.php]: Looks like is needed to propose a reset() method for this repetition.

File: src/Db/AbstractDb.php [hunk 4]
```diff
@@ -318,13 +330,27 @@ public function getSelect()
...
     }
+    
+    /**
+     * Prepares statement for Sql Object
+     * 
+     * @return StatementInterface
+     */
+    protected function getStatement()
+    {
+        if ($this->statement instanceof StatementInterface) {
```
Comment (Maks3w) [inline src/Db/AbstractDb.php]: Invert the logic here. If NOT then create and assign.

**Comment by weierophinney** [comment]
@andrey-mokhov We cannot merge this change without tests; will you have time to implement them?

**Comment by weierophinney** [comment]
Closing due to lack of updates.

**Pull request closed without merging**